promising/errors.py

```
@@ -89,3 +109,137 @@ class PromiseNotUnpackedError(PromiseInvalidStateError):
+        _previous_sys_excepthook(exc_type, exc_value, exc_tb)
+
+
+def _promising_threading_excepthook(args: threading.ExceptHookArgs) -> None:
+    if hasattr(args.exc_value, "__promising_context__"):
+        try:
+            _print_exception_with_promising_context(
+                args.exc_type,
+                args.exc_value,
+                args.exc_traceback,
+            )
+        except BaseException as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `BaseException`

(BLE001)